### ISSUE TITLE
Array option with default [] prints "[default: ]"

### ISSUE THREAD

**Issue by clux** [bcoe/yargs#132]
When using an array option that is optional (but should default to none), the help string will print this a little weirdly:

```js
var argv = require('yargs')
  .usage('Usage: $0 [options]')
  .options({
    'a': {
      alias: 'arryopt',
      describe: 'add perhaps many of these',
      array: true,
      default: []
    },
  })
  .help('h')
  .alias('h', 'help')
  .argv;
```

```sh
Usage: test.js [options]

Options:
  -a, --arryopt  add perhaps many of these                           [default: ]
  -h, --help     Show help
```

Which isn't really a big deal, but diminishes the value of the `default` option.
Using latest yargs version (3.6.0).

**Comment by clux** [comment]
Oh right. There's an option for this if I used `.default('a', [], 'string to print')`. Nevermind! Sorry :)

**Comment by bcoe** [comment]
I'd like an array to default to printing `[]`.

**Comment by bcoe** [comment]
@clux this should now be fixed in `yargs@3.7.0`,

`npm install yargs@3.7.0 --save`.

**Comment by clux** [comment]
Thanks, that is sensible :)
